pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/TableRebalanceIntegrationTest.java

```diff
@@ -1258,6 +1265,41 @@ private void checkRebalanceDryRunSummary(RebalanceResult rebalanceResult, Rebala
     assertEquals(numServersUnchanged, summaryResult.getServerInfo().getServersUnchanged().size());
   }
 
+  @Test
+  public void testDisallowMultipleConcurrentRebalancesOnSameTable() throws Exception {
+    // Manually write an IN_PROGRESS rebalance job to ZK instead of trying to collide multiple actual rebalance
+    // attempts which will be prone to race conditions and cause this test to be flaky. We only reject a rebalance job
+    // if there is an IN_PROGRESS rebalance job for the same table in ZK, so we could actually end up with more than
+    // one active rebalance job if both are started at the exact same time since the progress stats are written to ZK
+    // after some initial pre-checks are done. However, rebalances are idempotent, and we don't actually care too much
+    // about avoiding this edge case race condition as long as in most cases we are able to prevent users from
+    // triggering a rebalance for a table that already has an in-progress rebalance job.
+    String jobId = TableRebalancer.createUniqueRebalanceJobIdentifier();
+    String tableNameWithType = TableNameBuilder.forType(TableType.REALTIME).tableNameWithType(getTableName());
+    TableRebalanceProgressStats progressStats = new TableRebalanceProgressStats();
+    progressStats.setStatus(RebalanceResult.Status.IN_PROGRESS);
+    Map<String, String> jobMetadata = new HashMap<>();
+    jobMetadata.put(CommonConstants.ControllerJob.TABLE_NAME_WITH_TYPE, tableNameWithType);
+    jobMetadata.put(CommonConstants.ControllerJob.JOB_ID, jobId);
+    jobMetadata.put(CommonConstants.ControllerJob.SUBMISSION_TIME_MS, Long.toString(System.currentTimeMillis()));
+    jobMetadata.put(CommonConstants.ControllerJob.JOB_TYPE, ControllerJobType.TABLE_REBALANCE);
+    jobMetadata.put(RebalanceJobConstants.JOB_METADATA_KEY_REBALANCE_PROGRESS_STATS,
+        JsonUtils.objectToString(progressStats));
+    ControllerZkHelixUtils.addControllerJobToZK(_propertyStore, jobId, jobMetadata, ControllerJobType.TABLE_REBALANCE,
+        prevJobMetadata -> true);
+
+    RebalanceConfig rebalanceConfig = new RebalanceConfig();
+    String response = sendPostRequest(getRebalanceUrl(rebalanceConfig, TableType.REALTIME));
+    assertTrue(response.contains("Rebalance job is already in progress for table"));
```

Comment:
nit: can we also assert on the returned status code?

Reply:
have you tested this with the UI / swagger? How does that look with the `409 CONFLICT` changes?

Reply:
Looks like it's handled okay:

<img width="2541" alt="Screenshot 2025-06-10 at 21 57 13" src="https://github.com/user-attachments/assets/b0fb00e1-f959-4388-8cab-9a339bc61635" />

<img width="1407" alt="Screenshot 2025-06-10 at 21 57 53" src="https://github.com/user-attachments/assets/5b098008-d722-4f00-bbcd-19778db96eec" />